### DISCUSSION TITLE
can't get ps to run

### DISCUSSION THREAD

**Discussion by Davidhsherman13** [ChrisTitusTech/winutil#1864] [General]
.\winutil.ps1 : File C:\Users\dhs13\winutil.ps1 cannot be loaded. The file C:\Users\dhs13\winutil.ps1 is not digitally
signed. You cannot run this script on the current system. For more information about running scripts and setting
execution policy, see about_Execution_Policies at https:/go.microsoft.com/fwlink/?LinkID=135170.
At line:1 char:1
+ .\winutil.ps1
+ ~~~~~~~~~~~~~
    + CategoryInfo          : SecurityError: (:) [], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess

Any ideas?
 I download the *.ps1 today -4/22/2024

**Comment by og-mrk** [comment]
Hello @Davidhsherman13 
You can find a solution for your problem in the [know issues](https://github.com/ChrisTitusTech/winutil/blob/main/KnownIssues.md), in the last bullet point describing the issue and how to fix it.

Hope my answer helped you, If so, please go ahead and mark the discussion as answered. Thanks in advance 😄

**Reply by Davidhsherman13** [reply]
error message:
iex : At line:12832 char:5
+     class WingetFailedInstall : Exception {
+     ~~~~~
Class keyword is not allowed in ConstrainedLanguage mode.
At line:12832 char:11
+     class WingetFailedInstall : Exception {
+           ~~~~~~~~~~~~~~~~~~~
Unexpected token 'WingetFailedInstall' in expression or statement.
At line:12838 char:5
+     class ChocoFailedInstall : Exception {
+     ~~~~~
Class keyword is not allowed in ConstrainedLanguage mode.
At line:12838 char:11
+     class ChocoFailedInstall : Exception {
+           ~~~~~~~~~~~~~~~~~~
Unexpected token 'ChocoFailedInstall' in expression or statement.
At line:12844 char:5
+     class GenericException : Exception {
+     ~~~~~
Class keyword is not allowed in ConstrainedLanguage mode.
At line:12844 char:11
+     class GenericException : Exception {
+           ~~~~~~~~~~~~~~~~
Unexpected token 'GenericException' in expression or statement.
At line:1 char:26
+ irm christitus.com/win | iex
+                          ~~~
    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException
    + FullyQualifiedErrorId : ClassesNotAllowedInConstrainedLanguage,Microsoft.PowerShell.Commands.InvokeExpressionCom
   mand

**Reply by ginotru7362** [reply]
This is an older post, so I dont know if youll see this reply or if you found a solution?  I assume you ran the command as ADMIN?  To avoid issues with unsigned content im trying to run or install,  I run Windows with " Test Signing " mode enabled.  In CMD PROMPT or POWER SHELL with ADMIN privaleges, type:   bcdedit -set testsigning on  . With the newer versions and the heightened security Windows adopted, you might need to run:   bcdedit.exe  -set TESTSIGNING ON   . If you get a "permission denied" or error code, you might have to disable "secure boot" in the PCs BIOS. There are many guides  available with other fixes if this doesnt work for you. Hope this helps.  

[](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/the-testsigning-boot-configuration-option)

**Reply by soumilnegi154** [reply]
> This is an older post, so I dont know if youll see this reply or if you found a solution? I assume you ran the command as ADMIN? To avoid issues with unsigned content im trying to run or install, I run Windows with " Test Signing " mode enabled. In CMD PROMPT or POWER SHELL with ADMIN privaleges, type: bcdedit -set testsigning on . With the newer versions and the heightened security Windows adopted, you might need to run: bcdedit.exe -set TESTSIGNING ON . If you get a "permission denied" or error code, you might have to disable "secure boot" in the PCs BIOS. There are many guides available with other fixes if this doesnt work for you. Hope this helps.

you can check the `unblock` button in properties of the powershell script and it will run it
